server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregator.java

```diff
@@ -160,7 +159,7 @@ private AutoDateHistogramAggregator(
 
         DateHistogramAggregatorBridge bridge = new DateHistogramAggregatorBridge() {
             @Override
-            protected boolean canOptimize() {
+            protected boolean canOptimize(int subAggLength) {
                 return canOptimize(valuesSourceConfig, roundingInfos[0].rounding);
             }
```

Comment:
Taking `subAggLength` as input and not using it doesn't make lot of sense. There should be better way to abstract `canOptimize` method

Reply:
yep, the Aggregator class already have the subAggLength, removed this parameter from this interface.